alpha.py

```diff
@@ -54,14 +54,16 @@ def is_valid(self):
         TODO: implement value checks.
         """
         expected_types = {
```

Comment:
Since we have an exception (`self.alpha`) here, I suggest to rename `expected_types` to `expected`, which is more straight in natural language perspective.

Reply:
In the next version, we should check all conditions more carefully though.